### ISSUE TITLE
False positive CA1508 with 'using'

### ISSUE THREAD

**Issue by NN---** [dotnet/roslyn-analyzers#4433]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Do not use ReferenceEquals with value types`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

**Version**: v3.3.1 (Latest)

### Describe the bug

editorconfig
```
dotnet_diagnostic.ca1508.severity=warning
```

```cs
        class A : IDisposable { public void Dispose() { } }
        A? g()
        {
            return new A();
        }

        int f()
        {
            A? a = g();
            if (a == null) return 2;
            using (a) // CA1508
            {
                return 1;
            }
        }
```

### Expected behavior
No warning

### Actual behavior
Analyzer warning

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/4358.
